crates/red_knot_python_semantic/src/types.rs

```diff
@@ -415,6 +415,11 @@ impl<'db> Type<'db> {
             (_, Type::Unknown | Type::Any | Type::Todo) => false,
             (Type::Never, _) => true,
             (_, Type::Never) => false,
+            (Type::BooleanLiteral(_), Type::Instance(class))
```

Comment:
Nice catch, I really thought we had this already 🤦 

There are unit tests at the bottom of this module for `is_subtype_of`, ideally we'd add one for this case.